src/main/java/fr/moribus/imageonmap/commands/maptool/GiveCommand.java

```diff
@@ -36,109 +36,110 @@
-        }
-        /*} catch (InterruptedException | ExecutionException e) {
-            PluginLogger.warning(I.t("Can't access to mojang API to check the player UUID"));
-        }*/
-        return null;
-    }
-
     @Override
     protected void run() throws CommandException {
 
         if (args.length < 2) {
             throwInvalidArgument(I.t("You must give a valid player name and a map name."));
         }
 
-        final Player p = getPlayerParameter(0);
-
-        ImageMap map;
-        //TODO add support for map name with spaces "cool name" or name or "name" "cool name with a \" and some stuff"
-        // should work
-        OfflinePlayer player = null;
+        ArrayList<String> arguments = getArgs();
 
-        if (args.length < 4) {
-            if (args.length == 2) {
-                player = playerSender();
+        if (arguments.size() > 3) {
+            warning(I.t("Too many parameters! Usage: /maptool give <playername> [playerFrom]:<mapname>"));
+            return;
+        }
+        if (arguments.size() < 1) {
+            warning(I.t("Too few parameters! Usage: /maptool give <playername> [playerFrom]:<mapname>"));
+            return;
+        }
+        final String mapName;
+        final String from;
+        final String playerName;
+        final Player playerSender;
+        Player playerSender1;
+        try {
+            playerSender1 = playerSender();
+        } catch (CommandException ignored) {
+            if (arguments.size() == 2) {
+                throwInvalidArgument(I.t("Usage: /maptool give <playername> <playerFrom>:<mapname>"));
             }
-            if (args.length == 3) {
-                player = parse(2);
+            playerSender1 = null;
+        }
+        playerSender = playerSender1;
+        if (arguments.size() == 2) {
+            from = playerSender.getName();
+            playerName = arguments.get(0);
+            mapName = arguments.get(1);
+        } else {
+            if (arguments.size() == 3) {
+                from = arguments.get(1);
+                playerName = arguments.get(0);
+                mapName = arguments.get(2);
+            } else {
+                from = "";
+                playerName = "";
+                mapName = "";
+            }
+        }
+
+        //TODO passer en static
+        ImageOnMap.getPlugin().getCommandWorker().offlineNameFetch(from, uuid -> {
+            if (uuid == null) {
+                if (playerSender != null) {
+                    info(playerSender, I.t("The player {0} does not exist.", from));
+                }
+                return;
             }
-            map = MapManager.getMap(player.getUniqueId(), args[1]);
+            final ImageMap map = MapManager.getMap(uuid, mapName);
+
             if (map == null) {
-                throwInvalidArgument(I.t("Map not found"));
+                if (playerSender != null) {
+                    info(playerSender, I.t("This map does not exist."));
+                }
```

Comment:
`finalSender` blahblahblah
```suggestion
                warning(sender, I.t("This map does not exist."));
```